src/dotenv.ts

```diff
@@ -75,7 +75,8 @@ export async function loadDotenv(options: DotenvOptions): Promise<Env> {
   // Apply process.env
   if (!options.env?._applied) {
     Object.assign(environment, options.env);
-    environment._applied = true;
+    // @ts-expect-error environment variables are strings
+    options.env._applied = true;
```

Comment:
Can you please change it to assign a non enumarable key with empty string value (mainly for safety)